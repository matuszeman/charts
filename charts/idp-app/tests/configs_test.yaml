suite: configs
templates:
  - configs.yaml
  - deployment.yaml
values:
  - values/minimum.yaml
tests:
  - it: should match snapshot with values-ci.yaml
    values:
      - ../values-ci.yaml
    asserts:
      - matchSnapshot: {}
  - it: when fromConfigMap is specified, then do not generate managed ConfigMap and generate deployment volume
    values:
      - values/deployment.yaml
    set:
      containers:
        app:
          configDirs:
            not-managed-cm: /app-config
      configs:
        not-managed-cm:
          fromConfigMap: "{{ .Release.Name }}-not-managed-cm"
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
      - contains:
          path: spec.template.spec.volumes
          content:
            name: config-not-managed-cm
            configMap:
              name: RELEASE-NAME-not-managed-cm
        template: deployment.yaml

  - it: when config is specified and IS NOT used in container in as configFiles, then DO NOT generate deployment volume
    values:
      - values/deployment.yaml
    set:
#      containers:
#        app:
#          configFiles:
#            myconfig:
#              yes: /yes
      configs:
        myconfig:
          content:
            yes: no
    asserts:
      - equal:
          path: spec.template.spec.volumes
          value: null
        template: deployment.yaml
  - it: when config has both content and fromSecret or awsSecret specified, then use Secret instead of ConfigMap
    values:
      - values/deployment.yaml
    set:
      containers:
        app:
          configDirs:
            aws-secret: /config/aws-secret
            from-secret: /config/from-secret
          env:
            aws-secret-env:
              config:
                name: aws-secret
                key: key1
            from-secret-env:
              config:
                name: from-secret
                key: key1
      configs:
        aws-secret:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
          content:
            key1: value
        from-secret:
          fromSecret: "some-secret"
          content:
            key1: value
    asserts:
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: aws-secret-env
            valueFrom:
              secretKeyRef:
                key: key1
                name: RELEASE-NAME-config-aws-secret
        template: deployment.yaml
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: from-secret-env
            valueFrom:
              secretKeyRef:
                key: key1
                name: some-secret
        template: deployment.yaml
      - contains:
          path: spec.template.spec.volumes
          content:
            name: config-from-secret
            secret:
              secretName: some-secret
        template: deployment.yaml
      - contains:
          path: spec.template.spec.volumes
          content:
            name: config-aws-secret
            secret:
              secretName: RELEASE-NAME-config-aws-secret
        template: deployment.yaml
  - it: when config is specified and IS used in container, then DO generate deployment volume
    values:
      - values/deployment.yaml
    set:
      containers:
        app:
          configDirs:
            dirconfig: /config
          configFiles:
            fileconfig:
              yes: /yes
      configs:
        dirconfig:
          content:
            yes: no
        fileconfig:
          content:
            yes: no
    asserts:
      - contains:
          path: spec.template.spec.volumes
          any: true
          content:
            name: config-fileconfig
        template: deployment.yaml
      - contains:
          path: spec.template.spec.volumes
          any: true
          content:
            name: config-dirconfig
        template: deployment.yaml

  - it: when fromConfigMap and content is specified, then do not generate managed ConfigMap (prefer fromConfigMap)
    values:
      - values/deployment.yaml
    set:
      configs:
        not-managed-cm:
          fromConfigMap: "{{ .Release.Name }}-not-managed-cm"
          content:
            example.txt: example content
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
  - it: when fromSecret is specified, then do not generate managed Secret, generate deployment volume
    values:
      - values/deployment.yaml
    set:
      containers:
        app:
          configDirs:
            not-managed-secret: /app-config
      configs:
        not-managed-secret:
          fromSecret: "{{ .Release.Name }}-not-managed-secret"
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
      - contains:
          path: spec.template.spec.volumes
          content:
            name: config-not-managed-secret
            secret:
              secretName: RELEASE-NAME-not-managed-secret
        template: deployment.yaml
  - it: when fromSecret and content is specified, then do not generate managed Secret (prefer fromSecret)
    set:
      configs:
        not-managed-secret:
          fromSecret: "{{ .Release.Name }}-not-managed-secret"
          content:
            example.txt: example content
    asserts:
      - hasDocuments:
          count: 0
        template: configs.yaml
  - it: when awsSecret is specified, then generate SecretStore and ExternalSecret
    set:
      configs:
        aws-secrets:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
            refreshInterval: 1h
          content:
            example.txt: example content
    asserts:
      - matchSnapshot:
          template: configs.yaml
  - it: when global.idpAppConfig.defaults.configs.annotations is set
    set:
      global:
        idpAppConfig:
          defaults:
            configs:
              annotations:
                global-config-annotation: some-value
      configs:
        cm:
          content:
            some: data
        aws-secrets:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
    asserts:
      - matchSnapshot:
          template: configs.yaml
  - it: when awsSecret with versionId is specified, then set it to externalsecret
    set:
      configs:
        aws-secrets:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
            versionId: a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
    asserts:
      - equal:
          path: spec.dataFrom[0].extract.version
          value: uuid/a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
        template: configs.yaml
        documentIndex: 0
  - it: when awsSecret and valuesHash, then generate ExternalSecret annotation
    set:
      configs:
        aws-secrets:
          valuesHash: 1234aaa
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets-manual
    asserts:
      - equal:
          path: metadata.annotations.values-hash
          value: 1234aaa
        template: configs.yaml
        documentIndex: 0
  - it: when valuesHash and restartPodOnUpdate, then generate Deployment pod with config annotation
    values:
      - values/deployment.yaml
    set:
      podAnnotations:
        test: me
      configs:
        # this should not exist in annotations
        cm-default-do-not-restart:
          content:
            some: data
        cm-do-not-restart:
          restartPodOnUpdate: false
          content:
            some: data
        # below should be added to annotations
        cm:
          restartPodOnUpdate: true
          content:
            some: data
        from-cm:
          restartPodOnUpdate: true
          valuesHash: "abcd1111"
          fromConfigMap: "test-cm"
        from-secret:
          restartPodOnUpdate: true
          valuesHash: "abcd2222"
          fromSecret: "test-secret"
        aws-secret:
          restartPodOnUpdate: true
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets-manual
            valuesHash: 1234
        aws-secret-versioned:
          restartPodOnUpdate: true
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets-versioned
            versionId: versionXYZ
    asserts:
      - equal:
          path: spec.template.metadata.annotations
          value:
            config-hash-aws-secret: 0f845a68504ca3a4d93e2eee49a5735258ac101c
            config-hash-aws-secret-versioned: bc8f4130609e4f1944d560a4c16fdb8edef8a8ce
            config-hash-cm: b5f97e50a8d6fb167b83b3c7084b0d15109429c0
            config-hash-from-cm: abcd1111
            config-hash-from-secret: abcd2222
            test: me
        template: deployment.yaml
        documentIndex: 0
  - it: when aws secret with annotations
    set:
      configs:
        aws-secret:
          annotations:
            some: annotation
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
    asserts:
      - equal:
          path: metadata.annotations
          value:
            some: annotation
        template: configs.yaml
        documentIndex: 0
  - it: when aws secret with annotations and valuesHash
    set:
      configs:
        aws-secret:
          annotations:
            some: annotation
          valuesHash: fare4342
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
    asserts:
      - equal:
          path: metadata.annotations
          value:
            some: annotation
            values-hash: fare4342
        template: configs.yaml
        documentIndex: 0
  - it: when cm with annotations
    set:
      configs:
        cm:
          annotations:
            some: annotation
          content:
            data: 123
    asserts:
      - equal:
          path: metadata.annotations
          value:
            some: annotation
        template: configs.yaml
        documentIndex: 0
  - it: when awsSecret with no versionId is specified, then do not set it to externalsecret
    set:
      configs:
        aws-secrets:
          awsSecret:
            arn: arn:aws:iam::123456789012:role/external-secrets
            #versionId: a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
    asserts:
      - notExists:
          path: spec.dataFrom[0].extract.version
        template: configs.yaml
        documentIndex: 0