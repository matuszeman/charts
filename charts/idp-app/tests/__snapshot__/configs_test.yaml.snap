should match snapshot with values-ci.yaml:
  1: |
    apiVersion: v1
    data:
      file1: file1 content
      plaintextkey: sometext
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-app
  2: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-app-secrets
    spec:
      dataFrom:
        - extract:
            conversionStrategy: Default
            decodingStrategy: None
            key: arn:aws:iam::123456789012:role/external-secrets
            metadataPolicy: None
            version: uuid/a1b2c3d4-5678-90ab-cdef-EXAMPLE22222
      refreshInterval: 99999h
      secretStoreRef:
        kind: SecretStore
        name: RELEASE-NAME-resource-suffix-config-secretsmanager
      target:
        creationPolicy: Owner
        name: RELEASE-NAME-resource-suffix-config-app-secrets
        template:
          engineVersion: v2
          metadata:
            labels:
              app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
              app.kubernetes.io/managed-by: Helm
              app.kubernetes.io/name: app-name
              app.kubernetes.io/version: 0.1.0
  3: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        values-hash: 1234fdfdf
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-app-secrets-manual
    spec:
      dataFrom:
        - extract:
            conversionStrategy: Default
            decodingStrategy: None
            key: arn:aws:iam::123456789012:role/external-secrets-manual
            metadataPolicy: None
      refreshInterval: 99999h
      secretStoreRef:
        kind: SecretStore
        name: RELEASE-NAME-resource-suffix-config-secretsmanager
      target:
        creationPolicy: Owner
        name: RELEASE-NAME-resource-suffix-config-app-secrets-manual
        template:
          engineVersion: v2
          metadata:
            labels:
              app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
              app.kubernetes.io/managed-by: Helm
              app.kubernetes.io/name: app-name
              app.kubernetes.io/version: 0.1.0
  4: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-hashed-aws-secret-7a8f252
    spec:
      dataFrom:
        - extract:
            conversionStrategy: Default
            decodingStrategy: None
            key: arn:aws:iam::123456789012:role/external-secrets
            metadataPolicy: None
      refreshInterval: 99999h
      secretStoreRef:
        kind: SecretStore
        name: RELEASE-NAME-resource-suffix-config-secretsmanager
      target:
        creationPolicy: Owner
        name: RELEASE-NAME-resource-suffix-config-hashed-aws-secret-7a8f252
        template:
          engineVersion: v2
          metadata:
            labels:
              app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
              app.kubernetes.io/managed-by: Helm
              app.kubernetes.io/name: app-name
              app.kubernetes.io/version: 0.1.0
  5: |
    apiVersion: v1
    data:
      some: value
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-hashed-cm-2b8747e
  6: |
    apiVersion: v1
    data:
      nothing: none
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-not-mounted-as-volume-in-pod
  7: |
    apiVersion: v1
    data:
      php.ini: |
        memory_limit = 256M
        max_execution_time = 180
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-php
  8: |
    apiVersion: v1
    data:
      script.sh: |
        #!/bin/sh
        echo Works
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-scripts
  9: |
    apiVersion: external-secrets.io/v1beta1
    kind: SecretStore
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix-config-secretsmanager
    spec:
      provider:
        aws:
          auth:
            jwt:
              serviceAccountRef:
                name: RELEASE-NAME-resource-suffix
          region: eu-central-1
          service: SecretsManager
  10: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix
    spec:
      revisionHistoryLimit: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
          app.kubernetes.io/name: app-name
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
            app.kubernetes.io/name: app-name
        spec:
          affinity:
            podAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: security
                        operator: In
                        values:
                          - S1
                  topologyKey: topology.kubernetes.io/zone
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
                - podAffinityTerm:
                    labelSelector:
                      matchExpressions:
                        - key: security
                          operator: In
                          values:
                            - S2
                    topologyKey: topology.kubernetes.io/zone
                  weight: 100
          containers:
            - image: 123456.dkr.ecr.eu-central-1.amazonaws.com/idp-example-web-app-apache:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: apache
              name: apache
              ports:
                - containerPort: 80
                  name: apache
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: apache
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
            - args:
                - --quoted=arg1
                - --quoted arg2
              command:
                - customcmdquoted
              env:
                - name: configFromAwsSecret
                  valueFrom:
                    secretKeyRef:
                      key: somekey
                      name: RELEASE-NAME-resource-suffix-config-app-secrets
                - name: configFromConfigContent
                  valueFrom:
                    configMapKeyRef:
                      key: plaintextkey
                      name: RELEASE-NAME-resource-suffix-config-app
                - name: configFromConfigMap
                  valueFrom:
                    configMapKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-cm
                - name: configFromConfigMapHashed
                  valueFrom:
                    configMapKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-cm-hashed-4d4c3f1
                - name: configFromSecret
                  valueFrom:
                    secretKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-secret
                - name: configFromSecretHashed
                  valueFrom:
                    secretKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-secret-hashed-4aef4fd
                - name: configKeyTemplatedValue
                  valueFrom:
                    configMapKeyRef:
                      key: cm-key
                      name: app-config-RELEASE-NAME
                - name: configKeyValue
                  valueFrom:
                    configMapKeyRef:
                      key: cm-key
                      name: app-config
                - name: floatValue
                  value: "456.789"
                - name: intValue
                  value: "123"
                - name: secretKeyTemplatedValue
                  valueFrom:
                    secretKeyRef:
                      key: secret-key
                      name: app-secrets-RELEASE-NAME
                - name: secretKeyValue
                  valueFrom:
                    secretKeyRef:
                      key: some-key
                      name: app-secrets
                - name: stringValue
                  value: string
                - name: templatedValue
                  value: value-RELEASE-NAME
              image: 123456.dkr.ecr.eu-central-1.amazonaws.com/test-server:0.1.0
              imagePullPolicy: IfNotPresent
              lifecycle:
                preStop:
                  exec:
                    command:
                      - /bin/sh
                      - -c
                      - echo 'preStop hook'
              livenessProbe:
                httpGet:
                  path: /
                  port: app
              name: app
              ports:
                - containerPort: 9000
                  name: app
                  protocol: TCP
                - containerPort: 9000
                  name: anotherport
              readinessProbe:
                httpGet:
                  path: /
                  port: app
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /
                  port: app
                periodSeconds: 10
              volumeMounts:
                - mountPath: /app-config
                  name: config-app
                  readOnly: true
                - mountPath: /app-secrets
                  name: config-app-secrets
                  readOnly: true
                - mountPath: /app-secrets-manual
                  name: config-app-secrets-manual
                  readOnly: true
                - mountPath: /hashed-aws-secret
                  name: config-hashed-aws-secret
                  readOnly: true
                - mountPath: /hashed-cm
                  name: config-hashed-cm
                  readOnly: true
                - mountPath: /not-managed-secret
                  name: config-not-managed-secret
                  readOnly: true
                - mountPath: /scripts
                  name: config-scripts
                  readOnly: true
                - mountPath: /tmp/example.txt
                  name: config-not-managed-cm
                  readOnly: true
                  subPath: example.txt
                - mountPath: /etc/php.ini
                  name: config-php
                  readOnly: true
                  subPath: php.ini
                - mountPath: /app/public/var
                  name: ephemeral-app-cache
                  readOnly: false
                - mountPath: /app/public/readonly-by-default.yaml
                  name: some-files
                  readOnly: true
                  subPath: readonly-by-default.yaml
                - mountPath: /app/public/readonly.yaml
                  name: some-files
                  readOnly: true
                  subPath: readonly.yaml
                - mountPath: /app/public/writable.yaml
                  name: some-files
                  readOnly: false
                  subPath: writable.yaml
            - image: 123456.dkr.ecr.eu-central-1.amazonaws.com/logging:1.0.0
              imagePullPolicy: IfNotPresent
              name: logging
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
          enableServiceLinks: false
          nodeSelector:
            environment: test
          securityContext:
            sysctls:
              - name: net.ipv4.ip_unprivileged_port_start
                value: "0"
          serviceAccountName: RELEASE-NAME-resource-suffix
          tolerations:
            - effect: NoSchedule
              key: environment
              operator: Equal
              value: test
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
                  app.kubernetes.io/name: app-name
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: ScheduleAnyway
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
                  app.kubernetes.io/name: app-name
              maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: ScheduleAnyway
          volumes:
            - configMap:
                name: RELEASE-NAME-resource-suffix-config-app
              name: config-app
            - name: config-app-secrets
              secret:
                defaultMode: 256
                secretName: RELEASE-NAME-resource-suffix-config-app-secrets
            - name: config-app-secrets-manual
              secret:
                secretName: RELEASE-NAME-resource-suffix-config-app-secrets-manual
            - name: config-hashed-aws-secret
              secret:
                secretName: RELEASE-NAME-resource-suffix-config-hashed-aws-secret-7a8f252
            - configMap:
                name: RELEASE-NAME-resource-suffix-config-hashed-cm-2b8747e
              name: config-hashed-cm
            - configMap:
                name: RELEASE-NAME-not-managed-cm
              name: config-not-managed-cm
            - name: config-not-managed-secret
              secret:
                secretName: RELEASE-NAME-not-managed-secret
            - configMap:
                name: RELEASE-NAME-resource-suffix-config-php
              name: config-php
            - configMap:
                defaultMode: 511
                name: RELEASE-NAME-resource-suffix-config-scripts
              name: config-scripts
            - emptyDir:
                sizeLimit: 10Mi
              name: ephemeral-app-cache
            - name: some-files
              persistentVolumeClaim:
                claimName: pvc-files
when awsSecret is specified, then generate SecretStore and ExternalSecret:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-config-aws-secrets
    spec:
      dataFrom:
        - extract:
            conversionStrategy: Default
            decodingStrategy: None
            key: arn:aws:iam::123456789012:role/external-secrets
            metadataPolicy: None
      refreshInterval: 1h
      secretStoreRef:
        kind: SecretStore
        name: RELEASE-NAME-config-secretsmanager
      target:
        creationPolicy: Owner
        name: RELEASE-NAME-config-aws-secrets
        template:
          engineVersion: v2
          metadata:
            labels:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/managed-by: Helm
              app.kubernetes.io/name: app-name
              app.kubernetes.io/version: 0.1.0
  2: |
    apiVersion: external-secrets.io/v1beta1
    kind: SecretStore
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-config-secretsmanager
    spec:
      provider:
        aws:
          auth:
            jwt:
              serviceAccountRef:
                name: RELEASE-NAME
          region: eu-central-1
          service: SecretsManager
when global.idpAppConfig.defaults.configs.annotations is set:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        global-config-annotation: some-value
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-config-aws-secrets
    spec:
      dataFrom:
        - extract:
            conversionStrategy: Default
            decodingStrategy: None
            key: arn:aws:iam::123456789012:role/external-secrets
            metadataPolicy: None
      refreshInterval: 99999h
      secretStoreRef:
        kind: SecretStore
        name: RELEASE-NAME-config-secretsmanager
      target:
        creationPolicy: Owner
        name: RELEASE-NAME-config-aws-secrets
        template:
          engineVersion: v2
          metadata:
            labels:
              app.kubernetes.io/instance: RELEASE-NAME
              app.kubernetes.io/managed-by: Helm
              app.kubernetes.io/name: app-name
              app.kubernetes.io/version: 0.1.0
  2: |
    apiVersion: v1
    data:
      some: data
    kind: ConfigMap
    metadata:
      annotations:
        global-config-annotation: some-value
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-config-cm
  3: |
    apiVersion: external-secrets.io/v1beta1
    kind: SecretStore
    metadata:
      annotations:
        global-config-annotation: some-value
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-config-secretsmanager
    spec:
      provider:
        aws:
          auth:
            jwt:
              serviceAccountRef:
                name: RELEASE-NAME
          region: eu-central-1
          service: SecretsManager
